src/inference/src/ie_core.cpp

```diff
@@ -651,6 +651,12 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
             // as the result is being checked by the user
             strictly_check_dims = false;
         } else {
+            // check if Auto-Batch plugin registered
+            try {
+                GetCPPPluginByName("BATCH");
+            } catch (...) {
```

Comment:
maybe we need to catch only std::runtime_error https://github.com/openvinotoolkit/openvino/blob/master/src/common/util/src/os/lin/lin_shared_object_loader.cpp#L33? other exceptions can be re-thrown to users